test/test_binary_ufuncs.py

```diff
@@ -656,12 +640,23 @@ def to_np(value):
                 actual = base.clone()
                 # When base is a 0-dim cpu tensor and exp is a cuda tensor, we exp `pow` to work but `pow_` to fail, since
                 # `pow` will try to create the output tensor on a cuda device, but `pow_` needs to use the cpu tensor as the output
-                if base.dim() == 0 and base.device.type == 'cpu' and exponent.device.type == 'cuda':
-                    regex = 'Expected all tensors to be on the same device, but found at least two devices, cuda.* and cpu!'
+                if isinstance(exponent, torch.Tensor):
```

Comment:
Before this PR, all the tests using `_test_pow` were not using any inputs that would've caused an error due to broadcasting semantics.

Reply:
Since your search query is just a plain string you can just see if `error_message in str(e)`. However, I think we should more directly test for this case. That is, test that the exception is thrown for inputs for which you know it will throw with `self.assertRaisesRegex` and make sure that any other inputs in the test don't throw this exception. 

Also, is this documented behavior? @mruberry how do we usually handle inplace variants that don't support all the inputs that the normal variant supports (e.g. for pow broadcasting and bool inputs) ?

Reply:
I think that the [broadcasting semantics](https://numpy.org/doc/stable/user/basics.broadcasting.html#general-broadcasting-rules) are same as that of numpy, so if an inplace method invocation requires resizing the base tensor, then we don't proceed with it & throw an error.

It is documented [here](https://pytorch.org/docs/stable/notes/broadcasting.html#in-place-semantics) for PyTorch.